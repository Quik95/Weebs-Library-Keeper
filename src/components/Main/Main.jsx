import React, { useState, useEffect } from "react";

// MUI
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";

// styles
import useStyles from "./Main.styles.js";

// Custom components
import AnimeCard from "../AnimeCard/AnimeCard";

//hooks
import useBreakpoint from "../../hooks/useBreakpoint";

//helpers
import fetchSavedAnimed, {
  cancelFetchSavedAnime
} from "../../helpers/fetchSavedAnime";

export default function Main() {
  const classes = useStyles();
  const matchesXl = useBreakpoint("xl");
  const [animeList, setAnimeList] = useState([]);

  const fetchSavedAnimeAndSetState = async () => {
    const res = await fetchSavedAnimed();
    setAnimeList(res);
  };

  useEffect(() => {
    fetchSavedAnimeAndSetState();
    return () => cancelFetchSavedAnime();
  }, []);

  return (
    <main className={classes.main} data-testid="main">
      <Container maxWidth={matchesXl ? "xl" : "lg"}>
        <Grid container spacing={6}>
          {animeList.map(({ _id }) => (
            <AnimeCard key={_id} _id={_id} />
          ))}
        </Grid>
      </Container>
    </main>
  );
}
