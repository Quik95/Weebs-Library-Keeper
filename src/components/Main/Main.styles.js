import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  main: { marginTop: theme.spacing(2), padding: theme.spacing(0, 1) },
}));
