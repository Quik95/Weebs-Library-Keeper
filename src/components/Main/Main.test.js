import React from 'react';
import { render } from '@testing-library/react';

import Main from './Main';

describe('<Main/>', () => {
  const { getByTestId } = render(<Main />);
  it('Renders correctly', () => {
    const main = getByTestId('main');
    expect(main).toMatchSnapshot();
  });
});
