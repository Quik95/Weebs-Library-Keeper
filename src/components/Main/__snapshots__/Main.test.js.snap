// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> Renders correctly 1`] = `
<main
  className="makeStyles-main-1"
  data-testid="main"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={6}
    >
      <AnimeCard
        key="0"
      />
      <AnimeCard
        key="1"
      />
      <AnimeCard
        key="2"
      />
      <AnimeCard
        key="3"
      />
      <AnimeCard
        key="4"
      />
      <AnimeCard
        key="5"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</main>
`;
